src/generate_counterfactual.jl

```diff
@@ -79,16 +80,27 @@ function generate_counterfactual(
     )
 
     # Search:
-    timer = isnothing(timeout) ? nothing : Timer(timeout)
-    while !ce.search[:terminated]
-        update!(ce)
-        if !isnothing(timer)
-            yield()
-            if !isopen(timer)
-                @info "Counterfactual search timed out before convergence"
-                break
+    if isa(generator, GradientBasedGenerator)
+        timer = isnothing(timeout) ? nothing : Timer(timeout)
+        while !ce.search[:terminated]
+            update!(ce)
+            if !isnothing(timer)
+                yield()
+                if !isopen(timer)
+                    @info "Counterfactual search timed out before convergence"
+                    break
+                end
             end
         end
+
+    elseif isa(generator, HeuristicBasedGenerator)
+        if isa(M, Models.TreeModel)
+            new_x = Generators.feature_tweaking(ce.generator, ce.M, ce.x, ce.target)
+            ce.s′ = new_x
+            ce.search[:path] = [ce.search[:path]..., new_x]
+            ce.search[:terminated] = true
```

Comment:
Could you give a quick explanation to why these are set to true? seems counterintuitive but it obviously works i guess?